client/components/marketing-survey/cancel-jetpack-form/jetpack-cancellation-offer.tsx

```diff
@@ -0,0 +1,202 @@
+import {
+	PLAN_ANNUAL_PERIOD,
+	PLAN_MONTHLY_PERIOD,
+	PLAN_BIENNIAL_PERIOD,
+} from '@automattic/calypso-products';
+import { Button } from '@automattic/components';
+import formatCurrency from '@automattic/format-currency';
+import { localizeUrl } from '@automattic/i18n-utils';
+import { useTranslate } from 'i18n-calypso';
+import { useCallback, useMemo } from 'react';
+import FormattedHeader from 'calypso/components/formatted-header';
+import InlineSupportLink from 'calypso/components/inline-support-link';
+import JetpackLogo from 'calypso/components/jetpack-logo';
+import { CancellationOffer } from 'calypso/state/cancellation-offers/types';
+import type { Purchase } from 'calypso/lib/purchases/types';
+import type { FC } from 'react';
+
+interface Props {
+	purchase: Purchase;
+	siteId: number;
+	offer: CancellationOffer;
+	percentDiscount: number;
+}
+
+const JetpackCancellationOffer: FC< Props > = ( props ) => {
+	const { offer, purchase, percentDiscount } = props;
+	const translate = useTranslate();
+
+	const onClickAccept = useCallback( () => {
+		return;
+	}, [] );
+
+	const { offerHeadline, renewalCopy } = useMemo( () => {
+		const periods = offer.discountedPeriods;
+		const renewalPrice = formatCurrency( offer.rawPrice, offer.currencyCode );
+		const fullPrice = formatCurrency( offer.originalPrice, offer.currencyCode );
+		const headlineOptions = {
+			args: {
+				periods,
+				discount: percentDiscount,
+				name: purchase.productName,
+			},
+		};
+		const renewalCopyOptions = {
+			args: {
+				periods,
+				renewalPrice,
+				fullPrice,
+			},
+			components: {
+				strong: <strong />,
+			},
+		};
+
+		let offerHeadline;
+		let renewalCopy;
+
+		switch ( purchase.billPeriodDays ) {
+			case PLAN_BIENNIAL_PERIOD:
```

Comment:
Sharing something I've just learnt, in case you didn't know either: the `translate` function can be specified a [plural form](https://github.com/Automattic/wp-calypso/blob/7dfafccf2c723707cbf5819cd7fc2d8bb3703784/packages/i18n-calypso/README.md#translate-method). I'm not sure that would make things more readable here, though.

Reply:
I learned about this recently too!

The trouble I ran into here was that the singular and plural versions of the strings are required to have the same number of placeholders. Since I'm only using %(periods)s placeholder in the plural version of the strings, I had to do separate versions for singular and plural.